corehq/apps/users/views/mobile/users.py

```diff
@@ -122,8 +122,10 @@ def custom_data(self):
     def editable_user(self):
         try:
             user = CommCareUser.get_by_user_id(self.editable_user_id, self.domain)
-            if not user or user.is_deleted():
+            if not user:
                 raise Http404()
+            if user.is_deleted():
+                self.template_name = "users/deleted_account.html"
```

Comment:
nit: might be easier to reason about this if you overrode the `template_name` property to return what you want. otherwise this could be perceived as some magic.